orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/Execution.java

```diff
@@ -297,7 +297,7 @@ public Stage stageById(String stageId) {
   public Stage stageByRef(String refId) {
     return stages
       .stream()
-      .filter(it -> it.getRefId().equals(refId))
+      .filter(it -> refId.equals(it.getRefId()))
```

Comment:
Not sure we should be doing this if the method signature isn't `@Nonnull`

Reply:
I think I managed to NPE it with a unit test.